homeassistant/components/tplink/button.py

```diff
@@ -0,0 +1,91 @@
+"""Demo platform that offers a fake button entity."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Final
+
+from kasa import Device, Feature
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import (
+    CoordinatedTPLinkFeatureEntity,
+    TPLinkFeatureEntityDescription,
+    _description_for_feature,
+    _entities_for_device_and_its_children,
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class TPLinkButtonEntityDescription(
+    ButtonEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based sensor entity description."""
+
+
+BUTTON_DESCRIPTIONS: Final = [
+    TPLinkButtonEntityDescription(
+        key="test_alarm",
+    ),
+    TPLinkButtonEntityDescription(
+        key="stop_alarm",
+    ),
+]
```

Comment:
nit: Could be a tuple instead of a list

Reply:
Added to my todo. This was partially copied over from the base branch's sensor platform, so either I'll fix this on all PRs or leave it to a separate "polish" PR just beefore when we go through the merge-back-to-dev PR.

I suppose I could open up a draft PR to do that to have a place to keep a list of todos and track the status?